lib/ansible/modules/cloud/vmware/vca_vapp.py

```diff
@@ -229,6 +234,55 @@ def set_state(module):
         if not vapp.undeploy(action):
             module.fail('unable to undeploy vapp')
 
+def connect_to_network(module, vdc_name, vapp_name, network_name, network_mode):
+
+    nets = filter(lambda n: n.name == network_name, module.vca.get_networks(vdc_name))
+    assert len(nets) == 1
+    the_vdc = module.vca.get_vdc(vdc_name)
+    the_vapp = module.vca.get_vapp(the_vdc, vapp_name)
+    assert the_vapp
+    assert the_vapp.name == vapp_name
+
+    # Connect vApp
+    task = the_vapp.connect_to_network(nets[0].name, nets[0].href)
+    result = module.vca.block_until_completed(task)
+    assert result
+
+    # Connect VM
+    if(network_mode == 'pool'):
```

Comment:
@kaneda-fr so had a conversation about this with one of my internal vmware ppl and it seems to come from the fact that the image is attached to NAT, and so when the vapp gets spun up using that image, it gets NAT. We do this because we dont have an image per defined network.  This may behave differently with routed networks, but we dont have routed networks.

Some custom internal automation works around this by creating the vapp from the image, then editing the network on the vapp, then starting the vapp.  Since I dont see any way to edit the network on the vapp after its created here, that means this likely wont work for us for now.